src/test/gcp/firedata.spec.ts

```diff
@@ -0,0 +1,71 @@
+import * as nock from "nock";
+import {
+  APPHOSTING_TOS_ID,
+  APP_CHECK_TOS_ID,
+  GetTosStatusResponse,
+  getAcceptanceStatus,
+  getTosStatus,
+  isProductTosAccepted,
+} from "../../gcp/firedata";
+import { expect } from "chai";
+
+const SAMPLE_RESPONSE = {
+  perServiceStatus: [
+    {
+      tosId: "APP_CHECK",
+      serviceStatus: {
+        tos: {
+          id: "app_check",
+          tosId: "APP_CHECK",
+        },
+        status: "ACCEPTED",
+      },
+    },
+    {
+      tosId: "APP_HOSTING_TOS",
+      serviceStatus: {
+        tos: {
+          id: "app_hosting",
+          tosId: "APP_HOSTING_TOS",
+        },
+        status: "TERMS_UPDATED",
+      },
+    },
+  ],
+};
+
+describe("firedata", () => {
+  before(() => {
+    nock.disableNetConnect();
+  });
+  after(() => {
+    nock.cleanAll();
+    nock.enableNetConnect();
+  });
+
+  describe("getTosStatus", () => {
+    it("should return parsed GetTosStatusResponse", () => {
+      nock("https://mobilesdk-pa.googleapis.com")
+        .get("/v1/accessmanagement/tos:getStatus")
+        .reply(200, SAMPLE_RESPONSE);
+
+      expect(getTosStatus()).to.eventually.deep.equal(SAMPLE_RESPONSE as GetTosStatusResponse);
```

Comment:
You need to either return this or await it: https://www.chaijs.com/plugins/chai-as-promised/ - as is, it is not actually running this assertion.